main.tf

```diff
@@ -39,10 +39,24 @@ resource "aws_route53_zone" "zone" {
 locals {
   records_expanded = [
     for record in var.records : merge({
-      record_id = join("-", compact([lower(record.type), lower(record.name)]))
-      ttl       = null
-      records   = null
-      alias     = null
+      # if no name is set for the record, we use the zones name
+      record_id = join("-", compact([lower(record.type), try(lower(record.name), ""),
+        try(
+          "${lower(record.set_identifier)}-${lower(record.weight)}",                                    # weighted routing policy
+          "${lower(record.set_identifier)}-${lower(record.failover)}-${lower(record.health_check_id)}", # failover
+          null
+        ),
+      ]))
+
+      name            = ""
+      ttl             = null
+      records         = null
+      alias           = null
+      allow_overwrite = null
+      set_identifier  = null
+      weight          = null
+      failover        = null
+      health_check_id = null
```

Comment:
will be removed by merge anyway...

Reply:
are you sure? If I remove these lines it will break the current implementation ( see my other comments )

Reply:
terraform console shows that result.... that is weird..